source/base/parameter_handler.cc

```diff
@@ -612,6 +612,27 @@ namespace
           }
       }
   }
+
+  void
+  simplify_xml_recursively(boost::property_tree::ptree &source)
+  {
+    for (auto &p : source)
+      {
+        if (p.second.get_optional<std::string>("value"))
+          {
+            auto temp = p.second.get<std::string>("value");
+            p.second.clear();
+            p.second.put_value<std::string>(temp);
+          }
+        else if (p.second.get_optional<std::string>("alias"))
+          {}
+        else
+          {
+            // it must be a subsection
```

Comment:
Is there a way to assert this? I thought that there are also other fields (documentation, the pattern, ...?) and in any case it would be nice to have an assert here so that if we ever add other fields, we know that we need to update this place.